### ISSUE TITLE
Segmentation fault when dealing with JSON.stringify()

### ISSUE THREAD

**Issue by Implementist** [facebook/hermes#188]
Hi, there.
When I fuzzing hermes(version: 3ed8340) with the following test case it shows segmentation fault.
**Test Case:**
```
var NISLFuzzingFunc = function(value, replacer, space) {
    return JSON.stringify(value, replacer, space);
};

var NISLFuzzingParam0 = true;

var NISLFuzzingParam1 = function(b, c, d, e, f, g, h, k, l) {
    return {
        d1: b,
        d2: c,
        d3: d,
        d4: e,
        d5: f,
        d6: g,
        d7: h,
        d8: k,
        d9: l
    };
};

var NISLFuzzingParam2 = 1711499276.6637239937014953;

NISLFuzzingFunc(NISLFuzzingParam0, NISLFuzzingParam1, NISLFuzzingParam2);
```

**Stack Dump**
```
0.      Program arguments: ./hermes ./tc.js
1.      Program arguments: ./hermes ./tc.js
#0 0x0000000000496fc8 Hermes driver 0x496fc8 Hermes driver 0x49705b
#1 0x0000000000496fc8 Hermes driver 0x495089 Hermes driver 0x496a8b
#2 0x0000000000496fc8 __restore_rt__ (Hermes driver +0x496fc8)
#3 0x000000000049705b Hermes driver 0x8d578c Hermes driver 0x8d6355
...
#128 0x000000000049705b Hermes driver 0x810fe2
./hermes[0x496fc8]
...
Segmentation fault
```

**Comment by tmikov** [comment]
This is a very old version of Hermes. I don't think it reproduces with the current version.

**Comment by dulinriley** [comment]
When I run this on master right now on MacOS, I get this exception instead:
```
RangeError: Maximum nesting level in JSON stringifyer exceeded
    at stringify (native)
    at NISLFuzzingFunc (./script.js:2:26)
    at global (./script.js:23:16)
```

I believe I had fixed this a while ago in this commit: 2808d57a5d36d1d91e24262425dbdb4c5c5ba46f

Can you try using the most recent Hermes release version or master, either one should have the fix.
